pgoapi/poke_utils.py

```diff
@@ -1,11 +1,43 @@
 from __future__ import absolute_import
 import os
+from math import floor, sqrt
+from pgoapi.utilities import *
 from pgoapi.pokemon import Pokemon
 from pgoapi.game_master import PokemonData
 from pgoapi.protos.POGOProtos.Inventory import Item_pb2 as Enum_Items
 import csv
 
 
+#TCPM = CPM + ACPM
+#Stamina =  (BaseStamina + IndividualStamina) * TCPM
+#Attack = (BaseAttack + IndividualAttack) * TCPM
+#Defense = (BaseDefense + IndividualDefense) * TCPM
+#CP = MAX(10, FLOOR(Stamina0.5 * Attack * Def0.5 / 10))
+#ACPM = (Stamina/(BaseSta + IVSta)) - CPM
+def calcACPM(pokemon, pokemon_details):
+    if not all_in(['cp', 'cp_multiplier', 'individual_stamina', 'individual_attack', 'individual_defense'], pokemon):
+        return 0    
+
+    baseAttk = pokemon_details[str(pokemon['pokemon_id'])]['BaseAttack']
+    baseDef = pokemon_details[str(pokemon['pokemon_id'])]['BaseDefense']
+    baseStamina = pokemon_details[str(pokemon['pokemon_id'])]['BaseStamina']
+    cpm = pokemon['cp_multiplier']
+    return max(0, sqrt(sqrt( (100*pow(pokemon['cp'],2)) / ( pow((baseAttk + pokemon['individual_attack']), 2) * (baseDef + pokemon['individual_defense']) * (baseStamina + pokemon['individual_stamina'])  ) )) - cpm)
+
+def calcCP(pokemon, tcpm, pokemon_details):
```

Comment:
Well, I went with that line of thought that if it is not included the data is just zero and it fixed my max evo cp calculation on Pokemon that I incorrectly thought had missing data. Thanks!